src/hooks/useDeviceSize.js

```diff
@@ -0,0 +1,26 @@
+import { useEffect, useState } from 'react';
+
+const useMediaQuery = (query) => {
+  const [matches, setMatches] = useState(false);
+
+  useEffect(() => {
+    const media = window.matchMedia(query);
+    const updateMatch = () => setMatches(media.matches);
+    updateMatch();
+
+    media.addEventListener('change', updateMatch);
+    return () => media.removeEventListener('change', updateMatch);
+  }, [query]);
```

Comment:
`matches`가 변경될 때만 `useMemo`로 감싼 값을 재계산하여 불필요한 리렌더링을 막을 수 있습니다.
단, `useMemo`를 사용해도 `matches` 자체가 `useState`로 관리되기 때문에 리렌더링을 완전히 피할 수는 없습니다.
하지만 값이 변경되지 않으면 메모이제이션 값을 반환하므로 성능 이슈를 어느 정도 해결할 수 있습니다.

```suggestion
  useEffect(() => {
    const media = window.matchMedia(query);
    const updateMatch = () => setMatches(media.matches);
    updateMatch();

    media.addEventListener('change', updateMatch);
    return () => media.removeEventListener('change', updateMatch);
  }, [query]);

  return useMemo(() => matches, [matches]);
```

Reply:
먼저 좋은 피드백 감사합니다!! 어제 리렌더링 오류에 대해서 `memo`를 사용하면 최적화할 수 있다고 소개해주셨을때, `memo`만 추가하면 최적화가 다 되는데 일반적으로 왜 모든 컴포넌트에 사용되지 않을까?에 대해서 고민해봤습니다. 
그래서 `memo`와 `useMemo`에 대해서 조금 더 알아봤더니 리렌더링을 막기위해 이 2가지를 남용하는 것은 오히려 성능에 악영향을 미칠 수 있다고 하네요..! (관련 내용은 마지막에 링크로 첨부했습니다.)
그리고 제가 아직 `memo`와 `useMemo`에 대해 잘 모르기 때문에 메모이제이션에 관한 부분은 후원 상세페이지 연결 이후, 리팩토링 작업때 공부해서 적용해볼 예정입니다!
그때 다시 한 번 리뷰 부탁드리겠습니다. 감사합니다! 

https://f-lab.kr/insight/react-memo-optimization-20240619
https://f-lab.kr/insight/react-usememo-performance-optimization-20240627